src/pages/peers/PeerTable.tsx

```diff
@@ -27,28 +27,41 @@ interface Props {
   peers: Peer[] | null
 }
 
+interface PeerLatency {
+  rtt: string
+  loading: boolean
+}
+
+function getPingState(peerLatency: Record<string, PeerLatency>, peer: Peer): ReactElement {
+  if (peerLatency[peer.address]?.loading) return <CircularProgress size={20} />
+
+  if (peerLatency[peer.address]?.rtt) return <span>{peerLatency[peer.address]?.rtt}</span>
+
+  return <Autorenew />
+}
+
 function PeerTable(props: Props): ReactElement {
   const classes = useStyles()
   const { beeDebugApi } = useContext(SettingsContext)
 
-  const [peerLatency, setPeerLatency] = useState([{ peerId: '', rtt: '', loading: false }])
+  const [peerLatency, setPeerLatency] = useState<Record<string, PeerLatency>>({})
 
-  const PingPeer = (peerId: string) => {
-    setPeerLatency([...peerLatency, { peerId: peerId, rtt: '', loading: true }])
+  const pingPeer = (peerId: string) => {
+    setPeerLatency({ ...peerLatency, [peerId]: { rtt: '', loading: true } })
     beeDebugApi
       ?.pingPeer(peerId)
       .then(res => {
-        setPeerLatency([...peerLatency, { peerId: peerId, rtt: res.rtt, loading: false }])
+        setPeerLatency({ ...peerLatency, [peerId]: { rtt: res.rtt, loading: false } })
       })
       .catch(() => {
-        setPeerLatency([...peerLatency, { peerId: peerId, rtt: 'error', loading: false }])
+        setPeerLatency({ ...peerLatency, [peerId]: { rtt: 'error', loading: false } })
       })
```

Comment:
Hmm I am wondering if there is a race condition here. Not sure, but I'd probably use the arrow function for `setPeerLatency` which provides the previous value.